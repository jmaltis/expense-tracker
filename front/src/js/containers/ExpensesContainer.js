import React, {Component} from "react";
import ExpensesList from "../components/ExpensesList";
import {expensesAPI} from "../api/APIs";

export default class extends Component {
    constructor() {
        super();
        this.state = {
            expenses: []
        }
    }

    componentDidMount() {
        expensesAPI.getAll()
            // When success
            .then(content => {
                // Update state with the content
                this.setState({
                    expenses: content
                })
            });
    }

    render() {
        return (
            <ExpensesList expenses={this.state.expenses}/>
        )
    }
}