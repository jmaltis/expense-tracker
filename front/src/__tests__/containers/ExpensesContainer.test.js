import React from "react";
import ExpensesContainer from "../../js/containers/ExpensesContainer";
import renderer from "react-test-renderer";

test('ExpensesContainer should be rendered properly', () => {
    const rendered = renderer.create(
        <ExpensesContainer />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
});