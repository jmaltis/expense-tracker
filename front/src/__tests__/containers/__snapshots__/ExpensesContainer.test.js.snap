// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpensesContainer should be rendered properly 1`] = `
<div>
  <div
    className="expensesTitle"
  >
    <h3>
      My expenses
    </h3>
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Add expense
    </button>
  </div>
  <ul
    className="list-group"
  >
    <li
      className="list-group-item"
    >
      <div
        className="expense"
      >
        <span
          className="datetime"
        >
          2017-01-01T12:23:34
        </span>
        <span
          className="amount"
        >
          100
          $
        </span>
        <span
          className="description"
        >
          random description of expense
        </span>
      </div>
    </li>
  </ul>
</div>
`;
